zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSetting.java

```diff
@@ -190,10 +210,30 @@ void closeAndRemoveInterpreterGroup(String noteId) {
     }
   }
 
+  void closeAndRemoveInterpreterGroupByUser(String user) {
+    if (user.equals("anonymous")) {
+      user = "";
+    }
+    String processKey = getInterpreterProcessKey(user, "");
+    String sessionKey = getInterpreterSessionKey(user, "");
+    InterpreterGroup groupToRemove = null;
+    for (String intpKey : new HashSet<>(interpreterGroupRef.keySet())) {
```

Comment:
You might misunderstand the logic here. I iterate through the keyset and then judge whether the key string contains the processKey. It can not be done by Map.containsKey